rdr_service/dao/workbench_dao.py

```diff
@@ -452,8 +452,24 @@ def get_workspaces_with_user_detail(self, **kwargs):
         now = clock.CLOCK.now()
         sequest_hours_ago = now - timedelta(hours=sequest_hour)
         with self.session() as session:
+            subquery = (
+                session.query(distinct(WorkbenchWorkspaceUser.workspaceId))
+                    .filter(WorkbenchWorkspaceUser.isCreator == 1)
+                    .subquery()
+            )
             count_query = (session.query(distinct(WorkbenchWorkspaceApproved.workspaceSourceId))
-                           .filter(WorkbenchWorkspaceApproved.excludeFromPublicDirectory == 0,
+                           .join(WorkbenchWorkspaceUser, WorkbenchResearcher, WorkbenchInstitutionalAffiliations)
+                           .filter(WorkbenchWorkspaceUser.workspaceId == WorkbenchWorkspaceApproved.id,
+                                   WorkbenchWorkspaceUser.researcherId == WorkbenchResearcher.id,
+                                   WorkbenchInstitutionalAffiliations.researcherId == WorkbenchResearcher.id,
```

Comment:
Because of the set of foreign keys between `WorkbenchWorkspaceUser`, `WorkbenchWorkspaceApproved`, `WorkbenchResearcher`, and `WorkbenchInstitutionalAffiliations`, SqlAlchemy is already aware of how to join them all and is generating them in inner joins because of line 461. So the filters for `WorkbenchWorkspaceUser.workspaceId == WorkbenchWorkspaceApproved.id`, `WorkbenchWorkspaceUser.researcherId == WorkbenchResearcher.id` and `WorkbenchInstitutionalAffiliations.researcherId == WorkbenchResearcher.id` can be removed to help clarify the intent of the query.

Reply:
Wow, cool, I never know that. Done!